lib/logstash/outputs/dynatrace.rb

```diff
@@ -44,80 +68,234 @@ class Dynatrace < LogStash::Outputs::Base
       # Disable SSL validation by setting :verify_mode OpenSSL::SSL::VERIFY_NONE
       config :ssl_verify_none, validate: :boolean, default: false
 
-      default :codec, 'json'
+      # Include headers in debug logs when HTTP errors occur. Headers include sensitive data such as API tokens.
+      config :debug_include_headers, validate: :boolean, default: false
 
-      attr_accessor :uri, :plugin_version
+      # Include body in debug logs when HTTP errors occur. Body may be large and include sensitive data.
+      config :debug_include_body, validate: :boolean, default: false
 
       def register
-        @logger.debug("Registering plugin")
-        require 'net/https'
-        require 'uri'
-        @uri = URI.parse(@ingest_endpoint_url.uri.to_s)
-        @client = Net::HTTP.new(@uri.host, @uri.port)
-
-        if uri.scheme == 'https'
-          @client.use_ssl = true
-          @client.verify_mode = OpenSSL::SSL::VERIFY_NONE if @ssl_verify_none
+        # ssl_verification_mode config is from mixin but ssl_verify_none is our documented config
+        @ssl_verification_mode = 'none' if @ssl_verify_none
+
+        @ingest_endpoint_url = @ingest_endpoint_url.to_s
+
+        # TODO: I don't really understand how this mechanism works. Does it work?
+        # TODO try to remove this and see what happens
+        # We count outstanding requests with this queue
+        # This queue tracks the requests to create backpressure
+        # When this queue is empty no new requests may be sent,
+        # tokens must be added back by the client on success
+        @request_tokens = SizedQueue.new(@pool_max)
+        @pool_max.times { |_t| @request_tokens << true }
+        @requests = []
```

Comment:
I believe this is not doing anything and can be removed. See https://github.com/logstash-plugins/logstash-output-http/issues/139 for a more thorough explanation. Removing them does not break tests.